### PR TITLE
[OpenMP] Add optimization to remove the RPC client

### PR DESCRIPTION
**Pull request by jhuber6**
Summary:
Part of the work done in the `libc` project is to provide host services
for things like `printf` or `malloc`, or generally any syscall-like
behaviour. This scheme works by emitting an externally visible global
called `__llvm_libc_rpc_client` that the host runtime can pick up to get
a handle to the global memory associated with the client. We use the
presence of this symbol to indicate whether or not we need to run an RPC
server. Normally, this symbol is only present if something requiring an
RPC server was linked in, such as `printf`. However, if this call to
`printf` was subsequently optimizated out, the symbol would remain and
cannot be removed (rightfully so) because of its linkage. This patch
adds a special-case optimization to remove this symbol so we can
indicate that an RPC server is no longer needed.

This patch puts this logic in `OpenMPOpt` as the most readily available
place for it. In the future, we should think how to move this somewhere
more generic. Furthermore, we use a hard-coded runtime name (which isn't
uncommon given all the other magic symbol names). But it might be nice
to abstract that part away.

### CODE REVIEW

File: llvm/lib/Transforms/IPO/OpenMPOpt.cpp [hunk 1]
```diff
@@ -1507,6 +1510,40 @@ struct OpenMPOpt {
     return Changed;
   }
 
+  /// Tries to remove known runtime symbols that are optional.
+  bool removeRuntimeSymbols() {
```
Comment (jplehr) [inline llvm/lib/Transforms/IPO/OpenMPOpt.cpp:1514]: It's technically not "remove", right? Would that be more like `privatizeRpcClientGlobal` or so?

Comment (jhuber6) [reply llvm/lib/Transforms/IPO/OpenMPOpt.cpp:1515]: I was thinking we might want something for another global in the future. I updated the comment to make it clearer that  'remove' is referring to the symbol table.

Comment (jplehr) [reply llvm/lib/Transforms/IPO/OpenMPOpt.cpp:1515]: WFM

**Review by jplehr — APPROVED** [review]
LGTM

**Review by jdoerfert — APPROVED** [review]
LG

**Pull request merged**